### PR TITLE
Add support for building from git

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

**Comment by zyga** [comment]
This builds and boot tests okay.

File: recipes-support/snapd/snapd-2.58.3.inc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+SRC_URI = "	\
+	https://${GO_IMPORT}/releases/download/${PV}/snapd_${PV}.vendor.tar.xz	\
+	file://0001-mkversion-data-generate-supported-assert-formats-inf.patch \
```
Comment (bboozzoo) [inline recipes-support/snapd/snapd-2.58.3.inc:3]: this patch could likely be applied through `SRC_URI += ..` snapd.inc

File: recipes-support/snapd/snapd-2.58.3.inc [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+"
+
+SRC_URI[md5sum] = "852110945dd61f22258ae1a80af55447"
+SRC_URI[sha256sum] = "7b8319b5ce1c2957651d0fec8c935bfbee02a1340927d9055ac1bdfdb9c1fca5"
+
+LIC_FILES_CHKSUM = "file://${WORKDIR}/${PN}-${PV}/COPYING;md5=d32239bcb673463ab874e80d47fae504"
```
Comment (bboozzoo) [inline recipes-support/snapd/snapd-2.58.3.inc:9]: same here:
```
LIC_FILES_CHKSUM = "file://${S}/COPYING;md5=d32239bcb673463ab874e80d47fae504"
```

File: recipes-support/snapd/snapd-git.inc [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  git://github.com/snapcore/snapd.git;protocol=https;branch=release/2.58;destsuffix=git/ \
+  file://0001-mkversion-data-generate-supported-assert-formats-inf.patch \
+"
+
+PV = "2.58.3+git${SRCPV}"
+S = "${WORKDIR}/git"
+
+SRCREV = "6c067a778203ac0696084bf2b42bbf46ef35706a"
+
+LIC_FILES_CHKSUM = "file://${S}/COPYING;md5=d32239bcb673463ab874e80d47fae504"
```
Comment (bboozzoo) [inline recipes-support/snapd/snapd-git.inc:13]: and this can be dropped

File: recipes-support/snapd/snapd-2.58.3.inc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+SRC_URI = "	\
+	https://${GO_IMPORT}/releases/download/${PV}/snapd_${PV}.vendor.tar.xz	\
+	file://0001-mkversion-data-generate-supported-assert-formats-inf.patch \
+"
+
```
Comment (bboozzoo) [inline recipes-support/snapd/snapd-2.58.3.inc:5]: you can add this here:
```
S = "${WORKDIR}/${PN}-${PV}"
```
and drop if from snapd.inc

File: recipes-support/snapd/snapd-go.inc [hunk 5]
```diff
@@ -42,10 +43,10 @@ snapd_go_do_compile() {
 
 		# these *must* be built statically
 		for prog in snap-exec snap-update-ns snapctl; do
-			${GO} install -tags '${GO_BUILD_TAGS}' -mod=vendor -v \
+			${GO} install -tags '${GO_BUILD_TAGS}' -v \
 			      -ldflags="${GO_RPATH} -linkmode=external -extldflags '${HOST_CC_ARCH}${TOOLCHAIN_OPTIONS} ${GO_RPATH_LINK} ${LDFLAGS} -static'" \
 			      -trimpath \
-			      -mod=vendor \
+			      "${@bb.utils.contains('GOBUILDFLAGS', '-mod=vendor', '-mod=vendor', '', d)}" \
```
[the remark this answers is not in the document]
Comment (zyga) [reply recipes-support/snapd/snapd-go.inc:49]: Yeah.

I wish we had gotten to the bottom of that but let's postpone that fight :)

[the remark this answers is not in the document]
Comment (zyga) [reply recipes-support/snapd/snapd-go.inc]: Done

**Pull request merged**